weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CompareEvaluationsPage/compareEvaluationsContext.tsx

```diff
@@ -83,30 +77,32 @@ export const CompareEvaluationsProvider: React.FC<{
     }
     return {
       state: initialState.result,
-      setBaselineEvaluationCallId,
+      setSelectedCallIdsOrdered,
       setComparisonDimensions,
       setSelectedInputDigest,
       setSelectedMetrics,
       addEvaluationCall: (newCallId: string) => {
         const newEvaluationCallIds = [...evaluationCallIds, newCallId];
         setEvaluationCallIds(newEvaluationCallIds);
+        setSelectedCallIdsOrdered(newEvaluationCallIds);
```

Comment:
keep these separate, so that when you reorder them we don't re-query the db. We can also extend this to not re-query when an evaluation is removed, but this is complicated and out of scope for v0.

Reply:
seems a little weird to have both. i would instead create a helper hook that can take a list of ids to fetch which only reruns when the set changes. For example, you could re-order the list passed by the caller alphabetically, query, then return back in the original order. This way the component does not need to worry about maintaining two lists